File: c/examples/demo_readme.c
#include <emtrace/emtrace.h>

EMTRACE_MAGIC(uintptr_t)

int main(void) {
    emtrace_init();
    EMTRACELN("Hello World!");

    int a = 1;
    int b = 2;
    EMTRACELN_F("{} + {} = {}", VAL(int, a), VAL(int, b), VAL(int, a + b));
}

File: rust/examples/demo_readme.rs
use emtrace::{init, traceln};

fn main() {
    init(&mut std::io::stdout().lock()).unwrap();
    traceln!("Hello World!");

    let a = 1;
    let b = 2;
    traceln!("{} + {} = {}", i32: a, i32: b, i32: a + b);
}
